### ISSUE TITLE
✅ 🐞 Elementor Editor loads very slowly if Nested Elements are active and in use

### ISSUE THREAD

**Issue by bilalkhan891** [elementor/elementor#24076]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

I am using experimental features specifically Nested Elements. Since the last update to 3.16.6, I am facing the loading issue with only the page where I am using Tabs or Accordions. 

### Steps to reproduce

1. Enable "Nexted Elements".
2. Create two pages with and without Tabs and Accordions. (add decent amount of content if you don't see any issue)
3. Try to edit both the pages. 

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>Click to reveal</summary>

````txt


					
== Server Environment ==
	Operating System: Linux
	Software: LiteSpeed
	MySQL version: MariaDB Server v10.5.22
	PHP Version: 7.4.33
	PHP Memory Limit: 512M
	PHP Max Input Vars: 8000
	PHP Max Post Size: 128M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.3.2
	Site URL: https://staging.buildhd.com
	Home URL: https://staging.buildhd.com
	WP Multisite: No
	Max Upload Size: 128 MB
	Memory limit: 40M
	Max Memory limit: 512M
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.8.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/118.0.0.0 Safari/537.36 Edg/118.0.2088.46

== Active Plugins ==
	Crocoblock Wizard
		Version: 1.0.1
		Author: Crocoblock

	Duplicate Page
		Version: 4.5.3
		Author: mndpsingh287

	Elementor
		Version: 3.16.6
		Author: Elementor.com

	Elementor Pro
		Version: 3.16.2
		Author: Elementor.com

	Gravity Forms
		Version: 2.7.15
		Author: Gravity Forms

	JetEngine
		Version: 3.2.6
		Author: Crocoblock

	JetSmartFilters
		Version: 3.2.2.1
		Author: Crocoblock

	Performance Lab
		Version: 2.7.0
		Author: WordPress Performance Team

	UpdraftPlus - Backup/Restore
		Version: 1.23.10
		Author: UpdraftPlus.Com, DavidAnderson

	WordPress Importer
		Version: 0.8.1
		Author: wordpressdotorg

	Zoho Forms
		Version: 3.0.2
		Author: Zoho Forms


== Features ==
	Custom Fonts: 1
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Optimized DOM Output: Active by default
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Inactive by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active
	Upgrade Swiper Library: Active by default
	Grid Container: Inactive by default
	Default to New Theme Builder: Active by default
	Hello Theme Header & Footer: Active by default
	Editor Top Bar: Inactive by default
	Landing Pages: Active by default
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Global Style Guide: Inactive by default
	Page Transitions: Active by default
	Notes: Active by default
	Form Submissions: Active by default
	Scroll Snap: Active by default
	Menu: Active
	Taxonomy Filter: Inactive by default


== Log ==
	


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible

== Elementor Pro - Compatibility Tag ==
	

				
````

</details>

**Comment by bilalkhan891** [comment]
When I rolled back to the previous version of Elementor 3.16.5 and pro 3.16.2, it resolved the issue.

**Comment by thorsolutions** [comment]
+1

**Comment by SolunaCG** [comment]
+1

**Comment by bilalkhan891** [comment]
### Update on the issue:

After downgrading the Elementor plugins, the page loading issues are resolved now. Now, whenever I update anything in the Tabs layout, every thing inside it goes blank and it stays like that for a very longer while and come back.

**Comment by jodiroseduncan** [comment]
+1

**Comment by SolunaCG** [comment]
> ### Update on the issue:
> After downgrading the Elementor plugins, the page loading issues are resolved now. Now, whenever I update anything in the Tabs layout, every thing inside it goes blank and it stays like that for a very longer while and come back.

I can confirm this behaviour on my websites.

**Comment by Xavier1610** [comment]
I have the same issue ! Please let us know when this problem is solved and that we can safely update Elementor to the last version, thanks !

**Comment by nicholaszein** [comment]
Hello everyone!

Thank you for reporting and contributing to this issue. We are aware of this problem and we are currently investigating it.

We will update you as soon as we have more information.

@bilalkhan891 and @SolunaCG, could you please share with us what versions you have rolled back to?

Thanks

**Comment by SolunaCG** [comment]
I´m sorry for the misunderstanding and for me not being very clear. What I meant was that I can confirm the following on my websites:

> Now, whenever I update anything in the Tabs layout, every thing inside it goes blank and it stays like that for a very longer while and come back.

I´ve only tried it with Nested Tabs so far. I just tried it again and it took 4 minutes for me until the images in the galleries in the nested tabs were loaded. In the first gallery in Tab1 there are only 15 images and they´re all pretty small.

![tabs problem with galleries](https://github.com/elementor/elementor/assets/114682524/16954853-7b11-41b4-83a5-52a11e02f87f)

I just rolled back to Elementor 3.17.0 and Elementor Pro 3.17.0 to see if this solves the problem but the problem is still there for me.

**Comment by bilalkhan891** [comment]
> When I rolled back to the previous version of Elementor 3.16.5 and pro 3.16.2, it resolved the issue.

I have been able resolve the editor page not loading at all by rolling back to 3.16.2. That I also now see is fixed with the new 3.17.x versions. 

However, the loading of the tabs are still there.

Additionally, I noticed an issue with a page which was using the old version of the accordions. The page was not opening the Elementor Editor mode for me as well. (going into the loading screen and never resolve). I cannot remember how I resolved that and replaced the old accordions with the new one. 

I hope that helps.

**Comment by SolunaCG** [comment]
As a small addition to my previous post: 

I tried using the exact same "Nested Tabs" that I had on website 1 now on a different website 2. I used Elementors new "Cross domain copy/paste" and the content was inserted as expected. So that worked. 
What doesn´t work properly (again) is editing the page. After inserting the Nested Tabs the whole site became pretty much unusable! I can´t press "Update" and scrolling is very laggy. Even after reloading I cannot rlly work with that page cause everything is stuttering and the content of the Nested Tabs is still not loaded after about a minute now. 

The environments are completely different btw! Website 1 (the image that I shared in the earlier post) is from a clients website that´s hosted on webspace and the image here is on a testing environment on my big dedicated server where I can definitely say that it´s not something server spec related (php memory limit, etc.):

![tabs_not_working](https://github.com/elementor/elementor/assets/114682524/110db417-fd58-4f5c-82ef-17af9e0b1120)

I hope that helps :-)

**Comment by astrosnaut** [comment]
I am experiencing this issue on Mac Os, Chrome and Firefox with Nested Tabs. I am currently running Elementor 3.18.3. I tried doing all the things suggested by the normal support route to no avail. It does not appear to be a code or plugin conflict. It's making using the nested tabs feature not worth the time it takes to load and save.

**Comment by pumadog** [comment]
Still very slow for me as well on macOS Ventura with Safari and Chrome. Saving nested elements as a template leads to errors.

**Comment by computeralex** [comment]
Same problem on Mac OS Monterey w/ Brave, Chrome, and Safari.

**Comment by 79ho3ein** [comment]
I hope this problem will be solved soon. Nested elements are not really usable in intermediate-advanced designs and the editor gets completely out of control.

**Comment by lensbreak** [comment]
+1

**Comment by samuelmente** [comment]
+1 I currently have major issues with this on multiple sites.

**Comment by mvconline1** [comment]
+1

**Comment by Remarkable-Guille** [comment]
+2

**Comment by Jerostel** [comment]
+3

**Comment by JboyJW** [comment]
+1 Can't wait until Nested Elements are stable... awesome feature, but unbearably slow to edit, even in version 3.19.2

**Comment by debihazelden** [comment]
Same problem here +1

**Comment by krue81** [comment]
Same issue here... it is soo slow that is impossible to do any edits on the contents of the tabs without making the editor stall and crash...

**Comment by opatotto** [comment]
I think you can solve this problem with a different approach without errors or slow loading:

1.-Create a new page with only the tabs in this case no nested

2.-Create a template on Elementor selecting type Section for example Section-tab1, then create all the content you want in this section, I just copied all the containers I had on my nested tabs, using the same parameters of my container (if you have problems you can use Elementor safe mode if you wait too much like in my case). Repeat this process for every tab you want

3.- After you have your section templates you just need to add the text-type Shortcode something like this [elementor-template id="996"] in the tab you want to show the content.

4.-Check everything is working on different resolutions and adjust some configs on your tab element, and there you have it!

**Comment by k4znIm** [comment]
Yes, it can be solved this way, but it doesn't change the fact that the currently new nested tabs feature has a problem that needs to be fixed by Elementor.

**Comment by nicholaszein** [comment]
Hey everyone,

we have good news:

We are working on improving the performance across all nested elements, and the results are really promising! 💪 

There is still some infrastructure changes that need to be implemented for stability, and some other performance tweaks.

As soon as we have more updates we will let you know!

Kind regards

**Comment by sydavalon** [comment]
any updates?

**Comment by ivio-sarah** [comment]
Hoping for an update as well. I'm having the same problem and it got to where I couldn't even copy the accordion element to a template to edit. I asked Elementor support for help and they just told me to increase the server memory, which of course didn't work because that's not the problem, and then they told me they couldn't replicate the issue.

**Comment by romanklabal** [comment]
I had to resort to editing some of pages in Chrome, as Firefox simply freezes up and is unusable. Hoping for a Firefox fix soon.

**Comment by sydavalon** [comment]
> I think you can solve this problem with a different approach without errors or slow loading:
> 
> 1.-Create a new page with only the tabs in this case no nested
> 
> 2.-Create a template on Elementor selecting type Section for example Section-tab1, then create all the content you want in this section, I just copied all the containers I had on my nested tabs, using the same parameters of my container (if you have problems you can use Elementor safe mode if you wait too much like in my case). Repeat this process for every tab you want
> 
> 3.- After you have your section templates you just need to add the text-type Shortcode something like this [elementor-template id="996"] in the tab you want to show the content.
> 
> 4.-Check everything is working on different resolutions and adjust some configs on your tab element, and there you have it!

It worked for me. thx.

**Comment by m1ga** [comment]
@nicholaszein just to give an update here: WP 6.5, Elementor 3.22.0-latest-1712133255 (activated: optimized control loading and markup) and I have 16 accordion items with a container + image + headline + 3 or 4 text widgets + button in each of them.

It looks like this in a local page:

Firefox: all elements are loaded below the accordion (hidden), then after ~2min it will show all the content (still below the accordion) and then it will move it into the items (takes very very long).

Chrome: the same just takes ~10sec for all that (still too slow)

**Comment by nicholaszein** [comment]
Thanks for reporting this @m1ga! We are going to check it out.

Nested Elements are in the process of receiving significant performance improvements. However, not everything is ready for prime-time. The build you're using is very advanced and may not be stable. It's better to use either `3.21.0-cloud6` or `3.21.0-beta1`.

**Comment by m1ga** [comment]
3.21.0-beta1 doesn't have the `Nested Accordion` PRs merge so I don't need to test those. I'm fully aware of the stability parts as it is only a testing environment. Just wanted to test it with the current master improvements to see if it is going in the right direction yet. But it still needs some more adjustments to make it usable again.

**Comment by nicholaszein** [comment]
@m1ga,

It's best for now that you don't try to test these, as they are still in development and not even ready for `alpha` testing.

**Comment by dribgib** [comment]
@nicholaszein when do you anticipates the nested element performance fix? I've got about 100 installs that desperately need this :)

**Comment by nicholaszein** [comment]
@dribgib id say we're about 80% there. There are still some rough edges to smooth out, and a lot of testing to be done. I can't tell you an ETA because we're not ready yet to release these changes. What I can say is that the performance and stability improvements are very promising!

**Comment by k4znIm** [comment]
Thank you for sharing these information. It's really great to have feedback from you 👍

**Comment by AaronIO** [comment]
Just noting that this issue still persists... these nested elements can render a page virtually un-editable based on slow they become.

**Comment by midihead7** [comment]
@nicholaszein,

When editing a mega menu with nested elements on my staging site, Elementor 3.23.1 and Elementor Pro 3.23.0 provide no performance improvement at all. Editing a mega menu with nested elements is every bit as slow as before and the Update button just spins and never saves. memory usage in Chrome shows high memory usage at 1.1 GB while editing the header template that contains the meg a menu.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that the issue you raised was resolved with the release of Elementor v3.23.0! 🥳 

The time to interaction should improve by 60%-98%, and performance overall should be better. For a full breakdown please read [this article](https://developers.elementor.com/elementor-3-23-developers-update/#:~:text=Performant%20Nested%20Elements%20In%20the%20Editor).

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
